engines/router/missionctl/handlers/batch_http_handler.go

```diff
@@ -0,0 +1,124 @@
+package handlers
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"sync"
+
+	"github.com/gojek/turing/engines/router/missionctl"
+	"github.com/gojek/turing/engines/router/missionctl/errors"
+	"github.com/gojek/turing/engines/router/missionctl/log"
+	"github.com/gojek/turing/engines/router/missionctl/turingctx"
+)
+
+type batchHTTPHandler struct {
+	httpHandler
+}
+
+type batchResponse struct {
+	StatusCode int         `json:"code"`
+	ErrorMsg   string      `json:"error,omitempty"`
+	Data       interface{} `json:"data,omitempty"`
+}
+
+type batchResult struct {
+	BatchResult []batchResponse `json:"batch_result"`
+}
+
+// NewBatchHTTPHandler creates an instance of the Mission Control's prediction request handler
+func NewBatchHTTPHandler(mc missionctl.MissionControl) http.Handler {
+	return &batchHTTPHandler{httpHandler{mc}}
+}
+
+func (h *batchHTTPHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
+	var httpErr *errors.HTTPError
+	h.measureRequestDuration(httpErr)
+
+	// Create context from the request context
+	ctx := turingctx.NewTuringContext(req.Context())
+	// Create context logger
+	ctxLogger := log.WithContext(ctx)
+	defer func() {
+		_ = ctxLogger.Sync()
+	}()
+
+	// Get the unique turing request id from the context
+	turingReqID, err := turingctx.GetRequestID(ctx)
+	if err != nil {
+		ctxLogger.Errorf("Could not retrieve Turing Request ID from context: %v",
+			err.Error())
+	}
+	ctxLogger.Debugf("Received batch request for %v", turingReqID)
+
+	ctx = h.enableTracingSpan(ctx, req)
+
+	// Read the request body
+	requestBody, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		h.error(ctx, rw, errors.NewHTTPError(err))
+		return
+	}
+
+	//Valid request
+	if valid := json.Valid(requestBody); !valid {
```

Comment:
Is it necessary to explicitly check if the payload is a valid JSON or would it be sufficient to just capture a possible error from `json.Unmarshall` method below?

Reply:
My intention was to differential invalid json and missing `batch_request`, since it could be a valid json and fail in unmarshalling due to missing key. 

I'll removed the `batch_request` and this can be combined.